File: server/app/schemas/notifications.py
"""알림 도메인 Pydantic 스키마."""

from datetime import datetime

from pydantic import BaseModel, Field


class NotificationItem(BaseModel):
    notification_id: str
    type: str
    title: str
    body: str
    is_read: bool
    data: dict | None = None
    created_at: datetime


class NotificationListData(BaseModel):
    items: list[NotificationItem]
    unread_count: int = 0


# ── /home ─────────────────────────────────────────────────────────────────────
class HomeRoutineSummary(BaseModel):
    routine_id: str
    name: str
    next_day_label: str | None = None


class HomeData(BaseModel):
    user_name: str
    streak_days: int
    today_routine: HomeRoutineSummary | None = None
    upcoming_notifications: list[NotificationItem] = Field(default_factory=list)
    recent_volume_kg: float = 0.0

File: app/src/services/notifications.ts
import { apiFetch } from './api';

export interface NotificationItem {
  notification_id: string;
  type: string;
  title: string;
  body: string;
  is_read: boolean;
  data?: Record<string, unknown> | null;
  created_at: string;
}

export interface NotificationListData {
  items: NotificationItem[];
  unread_count: number;
}

export function getNotifications(token: string): Promise<NotificationListData> {
  return apiFetch<NotificationListData>('/api/v1/notifications', { token });
}

export function markNotificationRead(token: string, notification_id: string): Promise<NotificationItem> {
  return apiFetch<NotificationItem>(`/api/v1/notifications/${notification_id}/read`, {
    method: 'PATCH',
    token,
  });
}
